Libs/GObject/Core/Classes/Object.TypeDictionary.cs

```diff
@@ -0,0 +1,153 @@
+
+                // Search all System.Type which contain a [Wrapper(TypeName)]
+                // for the corresponding type.
+
+                // Possible Idea: Autogenerate a 'RegisterTypes.cs' file that
+                // on startup will add every type to the type dictionary?
+
+                // Quick Path: Find the first 'Word' in the type and lookup
+                // assemblies by that name. Do we hardcode references to 'Pango',
+                // 'Gtk', etc?
+
+                // foreach (var assembly in AppDomain.CurrentDomain.GetAssemblies())
+                // {
+                //     assembly.GetType()
+                // }
+
+                // Search through unloaded assemblies?
+
+                // TODO: For now, we'll just look through the type's inheritance chain
+                // and find the first already registered type (e.g. GtkWidget). Effectively,
+                // the lowest-common-denominator of functionality will be exposed.
+                while (!Contains(gtype))
+                {
+                    ulong parent = Sys.Methods.type_parent(gtype.Value);
+                    if (parent == 0)
+                        throw new Exception("Could not get Type from GType");
+
+                    // TODO: One-way registration?
+                    
+                    gtype = new Sys.Type(parent);
+                }
+
+                return gtypedict[gtype];
+            }
+
+            // Get GType from System.Type
+            internal static Sys.Type Get(System.Type type)
+            {
+                // Check Type Dictionary
+                if (typedict.TryGetValue(type, out var gtype))
+                    return gtype;
+
+                // If we are looking up a type that is not yet in
+                // the type dictionary, we are most likely registering
+                // a new type. Therefore, we should register the type
+                // and parent types recursively now to avoid having to
+                // do this in the future.
+                
+                // Retrieve the GType accordingly
+                if (IsSubclass(type))
+                {
+                    // We are a subclass
+                    // RegisterNativeType will recursively add this
+                    // and all parent types to the type dictionary
+                    RegisterNativeType(type);
+                    return typedict[type];
+                }
+                
+                // We are a wrapper, so register types recursively
+                Console.WriteLine("Registering Recursively");
+                System.Type baseType = type;
+                while (!Contains(baseType))
+                {
+                    Console.WriteLine(baseType.Name);
+                    var methodInfo = GetGTypeMethodInfo(baseType)!;
+                    var typeid = (ulong) methodInfo.Invoke(null, null);
+                    gtype = new Sys.Type(typeid);
```

Comment:
Why cast to `ulong` and reconstruct the GType ? If we are sure that a `GetGType()` method always return a `GObject.Type`, we can simply do:

```cs
gtype = (Type) methodInfo.Invoke(null, null);
```

Reply:
Yep, you're right, it should be like that. There was another step at one point which needed the typeid, I must've forgotten to refactor that out.

Reply:
Resolved in 0148472